### ISSUE TITLE
[Feature Request]Allow filters in select and cascader to be case insensitive

### ISSUE THREAD

**Issue by allochi** [iview/iview#2278]
### What problem does this feature solve?
the filter in select and cascader is a great feature, it allows users to find an item by typing either the full item label or part of it, unfortunately, it's case-sensitive, which means that users can also miss an item.

As an example, I'm building a database for my organisation to document natural disaster, in the list of hazards there are multiple types of "Tornado" items to select from, currently, the users can find these items if they type "Tornado", "ornado", but not "tornado".

### What does the proposed API look like?
maybe add a 'case-sensitive-filter' directive, which is set to true by default, and in case it's false, the filter predicate apply str.toLowerCase() on both side of the comparison.

**Comment by SergioCrisostomo** [comment]
@allochi I agree it should be possible to use `case-insensitive` search. But also in some use cases would be nice to search `a` and get results with `åâäã`. So I think we could add a `Function` to pass in the `filterable="myFilterFn"` instead. This would be most powerfull and customizable.
